.github/workflows/docs-preview.yml

```diff
@@ -0,0 +1,89 @@
+      },
+      "pull_request": {
+        "head_repo": "${{ github.event.pull_request.head.repo.full_name }}",
+        "head_sha":  "${{ github.event.pull_request.head.sha }}",
+        "number":    "${{ github.event.pull_request.number }}"
+      }
+    }
+
+# Grant `secrets.GITHUB_TOKEN` only the minimum permissions needed by actions to function:
+# https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token
+# NOTE: See the associated `preview` and `deploy` workflows called for when these permissions are needed.
+permissions:
+  contents: read
+  pull-requests: write
+
+# NOTE: Reusable workflows (`workflow_call`) default to empty secrets.
+#       Avoid using `secrets: inherit` in favor of passing only the required secrets to the job.
+jobs:
+  # WORKAROUND:
+  # The alternative is to copy/paste the JSON value to the `preview-context` input of each job.
+  #
+  # PROBLEM:
+  # - `<job-id>.with` restricts available contexts to only `github` and `needs`, it's not possible to use `env` context:
+  #   https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idwithinput_id
+  # - Likewise, workflows presently lack support for YAML anchors:
+  #   https://github.com/actions/runner/issues/1182
+  create-context:
+    name: 'Create Context'
+    runs-on: ubuntu-24.04
+    outputs:
+      preview-context: ${{ steps.set-preview-context.outputs.preview-context }}
+    steps:
+      - id: set-preview-context
+        # The output value must be a single-line; use `jq` to convert the JSON to minified:
+        # NOTE: YAML `>-` does not help fold this multi-line content due to use of indentation:
+        # https://github.com/orgs/community/discussions/26105#discussioncomment-3250413
+        run: echo "preview-context=$(jq --compact-output <<< "${PREVIEW_CONTEXT}")" >> "${GITHUB_OUTPUT}"
+
+  # The `prepare` job is for running steps in an untrusted context (necessary to build the docs):
+  # CAUTION: This runs a build script which the PR could modify for malicious purposes.
+  prepare:
+    needs: [create-context]
+    uses: docker-mailserver/docker-mailserver/.github/workflows/docs-preview-prepare.yml@main
+    with:
+      preview-context: ${{ needs.create-context.outputs.preview-context }}
+
+  # The `deploy` job is for running the remaining steps in a trusted context after building the PR branch:
+  # CAUTION: Do not execute any content from untrusted sources (the PR branch or the retrieved artifact from the `prepare` job)
+  deploy:
+    needs: [create-context, prepare]
+    uses: docker-mailserver/docker-mailserver/.github/workflows/docs-preview-deploy.yml@main
+    secrets:
+      NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
+      NETLIFY_SITE_ID:    ${{ secrets.NETLIFY_SITE_ID    }}
+    with:
+      preview-context: ${{ needs.create-context.outputs.preview-context }}
```

Comment:
Presently the `prepare` + `deploy` jobs here have `uses` with an explicit `<org>/<repo> ... @<branch>` value.

I could relax that to omit those. PRs for branches from forks would not run changes on those branches AFAIK so that'd be ok. Might be better for any future maintenance to run the PR change, but I'm not sure how that may affect PRs from forks, especially with `pull_request_target`, it'd be the workflow from the PR base branch (which most of the time would be `master`.

---

Which reminds me I forgot to change the branch here 😓 

```suggestion
  # The `prepare` job is for running steps in an untrusted context (necessary to build the docs):
  # CAUTION: This runs a build script which the PR could modify for malicious purposes.
  prepare:
    needs: [create-context]
    uses: docker-mailserver/docker-mailserver/.github/workflows/docs-preview-prepare.yml@master
    with:
      preview-context: ${{ needs.create-context.outputs.preview-context }}

  # The `deploy` job is for running the remaining steps in a trusted context after building the PR branch:
  # CAUTION: Do not execute any content from untrusted sources (the PR branch or the retrieved artifact from the `prepare` job)
  deploy:
    needs: [create-context, prepare]
    uses: docker-mailserver/docker-mailserver/.github/workflows/docs-preview-deploy.yml@master
    secrets:
      NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
      NETLIFY_SITE_ID:    ${{ secrets.NETLIFY_SITE_ID    }}
    with:
      preview-context: ${{ needs.create-context.outputs.preview-context }}
```